### ISSUE TITLE
text hidden in SQL Editor by a strange diagonal white rectangle

### ISSUE THREAD

**Issue by mor-lehr-deel** [dbeaver/dbeaver#22299]
### Description

See the image - half of the SQL Editor is hidden by a strange white diagonal rectangle.
This diagonal rectangle moves as I scroll the open script up or down.
This only happens after I open a connection to the database and the script contains at or after line 22 `SELECT * FROM pg_trigg`
This did not happen in previous versions.
ok:
![image](https://github.com/dbeaver/dbeaver/assets/151533558/74a1cae0-3d26-4432-98a5-cf7e512028fc)
if I add a few more newlines - the text is hidden by a strange white diagonal rectangle.
![image](https://github.com/dbeaver/dbeaver/assets/151533558/a26bace8-e0ad-4087-a8bd-81325f93e22b)

### DBeaver Version

Community 23.3.1

### Operating System

MacOs Sonoma 14.2.1

### Database and driver

Postgres 15.4, default drivers

### Steps to reproduce

1. Open a new SQL script
2. paste this script
```
SELECT 1;





















SELECT * FROM pg_trigg
```
4. Connect to the DB the script defaults
5. DBeaver "hangs" for 2-3 seconds and then this rectangle appears
6. Disconnect the connection - the triangle is gone and the text reappears

**Comment by E1izabeth** [comment]
Unfortunately, I can't reproduce the issue. Could you send such script as a file? Maybe some hidden symbols was accidentally added there.

**Comment by mdasu** [comment]
I have the same issue after upgrade.  

**DBeaver Version**
Community 23.3.1.202312241705

**Operating System**

MacOs Sonoma 14.1.1

**Database and driver**
Oracle / default drivers ojdbc11

**Steps to reproduce**
1. I just opened an existing script; nothing special done to reproduce.  

![image](https://github.com/dbeaver/dbeaver/assets/8571363/67b37cb6-8af4-4590-b65a-baadf1f2cc94)

**Comment by E1izabeth** [comment]
Please provide the log files. You can find instructions here https://dbeaver.com/docs/wiki/Log-files/.

**Comment by mdasu** [comment]
[Logs_Dbeaver.zip](https://github.com/dbeaver/dbeaver/files/13823546/Logs_Dbeaver.zip)

**Comment by Dazaman** [comment]
Having the same issue with DBeaver Community Edition `Version 23.3.1.202312241705` `Release Date: 25 December 2023`

Running on macOS Sonoma Version 14.2.1

![Screenshot 2024-01-04 at 3 49 27 pm](https://github.com/dbeaver/dbeaver/assets/13788151/fbe81ee4-8fb5-48c7-b63f-1473e3b98994)

**Comment by vadim-zakharyan** [comment]
Have the same issue in SQL Editor
it appears after query run attempt

**DBeaver Version**
Community 23.3.1

**Operating System**
MacOs Sonoma 14.0 (23A344)

**Database and driver**
Postgres 12.16, default drivers

**Logfiles:**
[DBeaverData.log.zip](https://github.com/dbeaver/dbeaver/files/13826445/DBeaverData.log.zip)

**Screenshot**
<img width="1052" alt="image" src="https://github.com/dbeaver/dbeaver/assets/55964508/659d6416-1f11-47f3-af9f-db175a680903">

**Comment by peterpk** [comment]
Seeing the same. DBeaver CE Version 23.3.2.202401072024 as installed by Homebrew on macOS 14.2.1.  ~~Resizing the query pane fixes it temporarily but it comes back.~~   Sorry, I lie, resizing the pane only moves the triangle around.

**Comment by clemlaugit** [comment]
i have the same issue.  for me it happens when i'm pressing up arrow to get to a different line, but then it only happens on certain queries.....

![image](https://github.com/dbeaver/dbeaver/assets/91501833/c631b627-98fe-4ac4-b5ba-47c44292d722)
![image](https://github.com/dbeaver/dbeaver/assets/91501833/b15d7fe2-2f47-4a1e-ad57-45f169ca627e)

![image](https://github.com/dbeaver/dbeaver/assets/91501833/0bd91ea1-a77a-4dce-969e-5c13cc6a9a32)

**Comment by serjiokov** [comment]
I tried to reproduce this behaviour on the latest OS version 14.2.1 but all works as expected for me.
<img width="1278" alt="image" src="https://github.com/dbeaver/dbeaver/assets/32031248/1d46c339-fcf9-467c-abb7-68481f4f3173">
I would recommend to use latest OS version.

**Comment by serjiokov** [comment]
For some test script content all looks fine 
<img width="1438" alt="image" src="https://github.com/dbeaver/dbeaver/assets/32031248/c1135455-ea05-416b-ad8f-5444e447907d">

**Comment by ryeldell** [comment]
this is an issue for me. I am on latest macos. Just showed up when upgrading to 23.3.2

**Comment by bdewilde** [comment]
+1 for me, same behavior as others have reported, on latest versions of macos and dbeaver. after a dbeaver upgrade, this triangle started happening.

**Comment by serjiokov** [comment]
@bdewilde Could you provide more details (script, table ddl, theme settings, chip, etc) because it is not reproducible on on my side.

**Comment by bdewilde** [comment]
Hi @serjiokov , this only seems to be an issue in scripts; scrolling in / resizing of that window causes the triangle to move around, but it appears on its own shortly after opening a new script. Interestingly, I can't get it to appear using the "dark" theme -- which I see you are using -- only the "light" and "classic" themes.

_Update:_ I stand corrected, I was able to get the triangle to appear in a script window with "dark" theme, it's just inconsistent and, somehow, ephemeral. So, a smaller problem.

**Comment by SDJayhawk** [comment]
I'm seeing the same thing. Here's an ultra simple script that is producing this blank space in the editor. I'm seeing this on 2 different MySQL databases, one local, one on AWS. 14-inch MacBook Pro, 2021, M1 Pro chip, 16 GB, Sonoma 14.2.1. DBeaver Community Version 23.3.3.202401211839

![Screenshot 2024-01-24 at 12 04 29 PM](https://github.com/dbeaver/dbeaver/assets/92556147/d3bf170a-70c8-4c5f-811e-586a393aec71)

create table test1 (
	id int auto_increment primary key,
	name varchar(100) not null,
	created timestamp not null default current_timestamp
);

create table test2 (
	id int auto_increment primary key,
	name varchar(100) not null,
	test1_id int null,
	created timestamp not null default current_timestamp
);

insert into test1 (name) 
select 'test1' as name 
union
select 'test2' as name 
union
select 'test3' as name 
union
select 'test4' as name 
union
select 'test5' as name 
union
select 'test6' as name 
union
select 'test7' as name 
union
select 'test8' as name 
union
select 'test9' as name ;

select n.id, p.wwwww 
from xxx.YYY n

**Comment by TobyHewing** [comment]
It appears to be something to do with the syntax validation. In my case the table name is being highlighted in red to suggest it is invalid (actually it isn't), as soon as I move to the next line the triangle appears. If Iput a space between my select &  from statements the syntax validation identifies the table name as valid (not highlighted in red) and moving the cursor does not cause the triangle to appear. It should be noted that depending on what row in the script I am on the triangle will either obscure half the screen or just me a small triangle on the right.

**Comment by uadfs089314** [comment]
Thank you @TobyHewing, disabling `Enable semantic analysis (experimental)` under Preferences > Editors > SQL Editor > Code Editor seems to fix the issue over here.

**Comment by serjiokov** [comment]
Same issue https://github.com/dbeaver/dbeaver/issues/22813

**Comment by ricson-chua** [comment]
Will this be addressed? Seems pointless to use an external client if this function does not work? 
Just to add you can still enable semantic analysis but Read database table/columns for semantic analsysis must be off. 

Still better to see objects color coded. 

> Thank you @TobyHewing, disabling `Enable semantic analysis (experimental)` under Preferences > Editors > SQL Editor > Code Editor seems to fix the issue over here.

**Comment by E1izabeth** [comment]
Unfortunately, we can't reproduce it. It looks like some mac-specific rendering issue and we're unsure, what could cause it, to fix the bug blindly. We keep this issue in mind, but haven't figured out a solution yet.

**Comment by afreller-1wa** [comment]
I can confirm disabling "Enable semantic analysis (experimental)" in Code Editor preferences does stop the behaviour.  

OR

just disabling "Read database table/columns for semantic analsysis"
![Screenshot 2024-02-21 at 8 53 28 AM](https://github.com/dbeaver/dbeaver/assets/50186072/36ec2326-4292-4e32-843f-26cbff13945e)

**Comment by afreller-1wa** [comment]
I was wondering, could that be an Apple Silicon issue or do all macs have the same issue?

**Comment by peterpk** [comment]
> I was wondering, could that be an Apple Silicon issue or do all macs have the same issue?

I have an Intel Mac and I see the same issue.

**Comment by z0dijak** [comment]
Thanks for the hint @afreller-1wa . Just disabling "Read database table/columns for semantic analsysis" definitely helped to remove the white triangle.

I am using an M1 Mac on the latest MacOS + latest DBeaver version.

**Comment by jean-francois-labbe** [comment]
Also facing this issue on:
**dbeaver** 23.3.5
**macos** 14.3.1

Disabling "Read database table/columns for semantic analysis" fixies it.

In my case it seems that the "g" letter is triggering the behavior.

**Here is one that is triggering the triangle shape.**
**You need to move the result tab up to trigger the triangle.**
```sql
-- lorem ipsum
-- lorem ipsum
-- lorem ipsum
-- lorem ipsum
-- lorem ipsum
-- lorem ipsum
-- lorem ipsum
-- lorem ipsum
SELECT 1,  g
-- lorem ipsum
-- lorem ipsum
-- lorem ipsum
-- lorem ipsum
-- lorem ipsum
```

Here is a GIF demonstrating the issue with the snippet above. @E1izabeth can you reproduce with this snippet ?
![CleanShot 2024-02-29 at 11 52 18](https://github.com/dbeaver/dbeaver/assets/1558372/942983f9-1ac3-4742-aaa5-baff98b79fa2)

**Comment by E1izabeth** [comment]
@jean-francois-labbe thank you, we'll try

**Comment by meglio** [comment]
Just for reference, I updated to `Version 24.0.2.202404071654` and the issue is still there (MacOS).

**Comment by E1izabeth** [comment]
@meglio the fix will be available in the next version 24.0.3
